evm_arithmetization/src/witness/transition.rs

```diff
@@ -276,41 +331,64 @@ fn perform_op<F: Field>(
         }
         Operation::BinaryArithmetic(op) => generate_binary_arithmetic_op(op, state, row)?,
         Operation::TernaryArithmetic(op) => generate_ternary_arithmetic_op(op, state, row)?,
-        Operation::KeccakGeneral => generate_keccak_general(state, row)?,
+        Operation::KeccakGeneral => {
+            generate_keccak_general(state, row, is_interpreter, &preinitialized_segments)?
+        }
         Operation::ProverInput => generate_prover_input(state, row)?,
         Operation::Pop => generate_pop(state, row)?,
-        Operation::Jump => generate_jump(state, row)?,
-        Operation::Jumpi => generate_jumpi(state, row)?,
+        Operation::Jump => generate_jump(any_state, row, is_jumpdest_analysis)?,
+        Operation::Jumpi => generate_jumpi(any_state, row, is_jumpdest_analysis)?,
         Operation::Pc => generate_pc(state, row)?,
         Operation::Jumpdest => generate_jumpdest(state, row)?,
         Operation::GetContext => generate_get_context(state, row)?,
         Operation::SetContext => generate_set_context(state, row)?,
-        Operation::Mload32Bytes => generate_mload_32bytes(state, row)?,
+        Operation::Mload32Bytes => {
+            generate_mload_32bytes(state, row, is_interpreter, &preinitialized_segments)?
+        }
         Operation::Mstore32Bytes(n) => generate_mstore_32bytes(n, state, row)?,
         Operation::ExitKernel => generate_exit_kernel(state, row)?,
-        Operation::MloadGeneral => generate_mload_general(state, row)?,
+        Operation::MloadGeneral => {
+            generate_mload_general(state, row, is_interpreter, &preinitialized_segments)?
+        }
         Operation::MstoreGeneral => generate_mstore_general(state, row)?,
     };
+    match any_state {
+        State::Generation(state) => {}
+        State::Interpreter(interpreter) => {
+            interpreter.clear_traces();
```

Comment:
This `clear_traces` method is kinda ugly IMO.

Reply:
I guess is necessary unless the "log" logic is separated from the state change logic in all the `generate_*`  methods

Reply:
Indeed, but I am changing all `push` methods but `push_memory` to only push the operation if we are dealing with a `GenerationState`. This would address this comment and also your comments @4l0n50.